h2o-r/h2o-package/R/frame.R

```diff
@@ -3228,34 +3230,53 @@ as.h2o.Matrix <- function(x, destination_frame="", ...) {
   .key.validate(destination_frame)
   if ( destination_frame=="" ) # .h2o.readSVMLight wont handle ""
     destination_frame <- .key.make("Matrix") # only used if `x` variable name not valid key
-  
-  tmpf <- tempfile(fileext = ".svm")
-  .h2o.write.matrix.svmlight(x, file = tmpf)
-  h2f <- .h2o.readSVMLight(tmpf, destination_frame = destination_frame)
-  file.remove(tmpf)
-  h2f
+
+  drs = as.simple_triplet_matrix(x)# need to convert sparse matrix x to a simple triplet matrix format
+  thefile <- tempfile()
+  .h2o.write_stm_svm(drs, file = thefile)
+  h2f <- h2o.uploadFile(thefile, parse_type = "SVMLight")
+  unlink(thefile)
+  h2f[- c(1)]# remove the first column
+
 }
 
-.h2o.write.matrix.svmlight <- function(matrix, file) {
-  on.exit(sink())
-  sink(file)
-  sapply(1:nrow(matrix), function(i) {
-    r <- matrix[i, ]
-    val.indices <- which(r != 0)
-    val.indices <- val.indices[val.indices > 1]
-    target <- r[1]
-    features <- paste(sprintf("%d", val.indices - 1), r[val.indices], collapse = " ", sep = ":")
-    line <- sprintf("%s %s\n", target, features)
-    cat(line)
-  })
+#' Convert a simple triplet matrix to svm format
```

Comment:
Hidden functions don't need roxygen headers.  Let's remove those for consistency.  If there are notes about the function and what it should produce, you can put that as a comment in the function itself.